### PR TITLE
Change exception type/message when project item does not exist

### PR DESCRIPTION
**Pull request by maxbanas**
This is related to the discussion in #378  
`RazorSourceGenerator.GenerateCodeAsync(RazorLightProjectItem projectItem)` and `RazorSourceGenerator.CreateCodeDocumentAsync(RazorLightProjectItem projectItem)` both accept a `RazorLightProjectItem` with the precondition that the `.Exists` property is true. Previously, when this precondition was not met, `GenerateCodeAsync` threw a `TemplateNotFoundException` while `CreateCodeDoucmentAsync` threw a `InvalidOperationException`.  The exception language and type (in `GenerateCodeAsync `) made it sound like these methods are trying to "look up" a template, when in reality they are only checking the `Exists` property on the project item. These changes clarify the behavior of these methods and make the exception type consistent.

### CODE REVIEW

**Comment by maxbanas** [comment]
Tests passed for netcoreapp2.0 and netcoreapp3.1, but I am not able to run the tests for net5.0 yet.

File: src/RazorLight/Generation/RazorSourceGenerator.cs [hunk 1]
```diff
@@ -108,7 +108,7 @@ public virtual async Task<RazorCodeDocument> CreateCodeDocumentAsync(RazorLightP
 
 			if (!projectItem.Exists)
 			{
-				throw new InvalidOperationException($"Project can not find template with key {projectItem.Key}");
+				throw new InvalidOperationException($"RazorLightProjectItem with key {projectItem.Key} must exist");
```
Comment (jzabroski) [inline src/RazorLight/Generation/RazorSourceGenerator.cs:111]: Use `$"nameof(RazorLightProjectItem) of type projectItem.GetType().FullName with key {projectItem.Key} does not exist."

My rationale being that it would help when people ask for help to know exactly what type of RazorLightProjectItem they passed in.

File: src/RazorLight/Generation/RazorSourceGenerator.cs [hunk 2]
```diff
@@ -71,7 +71,7 @@ public async Task<IGeneratedRazorTemplate> GenerateCodeAsync(RazorLightProjectIt
 
 			if (!projectItem.Exists)
 			{
-				throw new TemplateNotFoundException($"Project can not find template with key {projectItem.Key}");
+				throw new InvalidOperationException($"RazorLightProjectItem with key {projectItem.Key} must exist");
```
Comment (jzabroski) [inline src/RazorLight/Generation/RazorSourceGenerator.cs:74]: Use `$"nameof(RazorLightProjectItem) of type projectItem.GetType().FullName with key {projectItem.Key} does not exist."

My rationale being that it would help when people ask for help to know exactly what type of RazorLightProjectItem they passed in.

**Review by jzabroski — CHANGES_REQUESTED** [review]

File: src/RazorLight/Generation/RazorSourceGenerator.cs (hunk 2 above)
Comment (maxbanas) [reply src/RazorLight/Generation/RazorSourceGenerator.cs]: I've made the requested changes. I also added two additional tests since the new exception messages require that projectItem is not null.

File: src/RazorLight/Generation/RazorSourceGenerator.cs (hunk 1 above)
Comment (maxbanas) [reply src/RazorLight/Generation/RazorSourceGenerator.cs]: I've made the requested changes. I also added two additional tests since the new exception messages require that projectItem is not null.

**Review by jzabroski — APPROVED** [review]

**Comment by jzabroski** [comment]
Thanks.  I will try to push this out tonight.
